sklearn/mixture/tests/test_gmm.py

```diff
@@ -157,12 +159,12 @@ def test_GMM_attributes():
     covars = (0.1 + 2 * rng.rand(n_components, n_features)) ** 2
     g.covars_ = covars
     assert_array_almost_equal(g.covars_, covars)
-    assert_raises(ValueError, g._set_covars, [])
-    assert_raises(ValueError, g._set_covars,
-                  np.zeros((n_components - 2, n_features)))
-
-    assert_raises(ValueError, mixture.GMM, n_components=20,
-                  covariance_type='badcovariance_type')
+    with pytest.raises(ValueError):
```

Comment:
Why getting rid of ``assert_raises`` here? I can't find any explanation. Also, I think there are many other places using ``assert_raises``. Maybe revert it and open an issue if needed?

Reply:
You are right - better be systematic. Reverted this.